File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.HashSet;
import java.util.Set;
    
class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        // 1. HashSet을 선언한다.
        Set<String> set = new HashSet<>();
        
        // 2. 모든 전화번호를 HashMap에 넣는다.
        for (String phone : phone_book){
            set.add(phone);
        }
        
        for (String phone : phone_book) {
            for(int i=0; i<phone.length(); i++){
                if(set.contains(phone.substring(0,i))){
                    answer = false;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution1(phone_book):
    answer = True
    phone_book.sort()
    for i in range(len(phone_book)-1):
        if phone_book[i+1][:len(phone_book[i])] == phone_book[i]:
            answer = False
    return answer

















def solution(phone_book):
    phone_book.sort()
    for i in range(len(phone_book)-1):
        if phone_book[i+1][:len(phone_book[i])] == phone_book[i]:
            return False
    return True



